### PR TITLE
add missing docs (external scheduler and plugin extender)

### LINKED ISSUE #285: PluginExtender documentation
add the doc with some example

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/area simulator
/kind documentation

#### Which issue(s) this PR fixes:

Fixes #285 

#### Special notes for your reviewer:

Some parts are using the implementation of https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/288. 🙏 

/priority next-release
/label tide/merge-method-squash

### CODE REVIEW

**Comment by sanposhiho** [comment]
Ignore the test failing, that's because of:
> Some parts are using the implementation of https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/288. 🙏

**Comment by sanposhiho** [comment]
/hold

https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/288 should be merged first.

File: simulator/docs/external-scheduler.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+## the external scheduler
```
Comment (196Ikuchil) [inline simulator/docs/external-scheduler.md:1]:
```suggestion
## The external scheduler
```

File: simulator/docs/external-scheduler.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+This document describes how to use the external scheduler instead of the scheduler running in the simulator.
+We use the [`externalscheduler` package](../pkg/externalscheduler).
+
```
Comment (196Ikuchil) [inline simulator/docs/external-scheduler.md:5]: It is helpful for the user to specify at the beginning "what the user will be able to do" with this setting to help them decide whether to use it or not.

File: simulator/docs/external-scheduler.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+As you see, `externalscheduler.NewSchedulerCommand` has much similar interface as the `app.NewSchedulerCommand`.
+You can register your plugins by `externalscheduler.WithPlugin` option.
+
+Via this step, all Pods scheduled by this scheduler will get the scheduling results in the annotation like in the simulator!
+
+### [Optional] Connect your scheduler to the kube-apiserver in the simulator
+
+If you want to connect your scheduler in your cluster's kube-apiserver, skip this. 
+(But, note that it, of course, means your Pods on your cluster will be actually scheduled by the scheduler)
+
```
Comment (196Ikuchil) [inline simulator/docs/external-scheduler.md:52]: > It is helpful for the user to specify at the beginning "what the user will be able to do" with this setting to help them decide whether to use it or not.

Also here. We will be able to use webUI?

**Review by 196Ikuchil — COMMENTED** [review]
@sanposhiho 
Left some comments and proposal. Plz check it🙏

File: simulator/docs/external-scheduler.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+This document describes how to use the external scheduler instead of the scheduler running in the simulator.
+We use the [`externalscheduler` package](../pkg/externalscheduler).
+
+Note that the scheduler built with the [`externalscheduler` package](../pkg/externalscheduler) will work much like the normal scheduler.
+But, it does some additional effort inside to export the scheduling results. So, we'd recommend using it in the dev env only.
+
+### Change your scheduler
+
+Here, we assume you're registering your custom plugins in your scheduler like this:
```
[the remark this answers is not in the document]
Comment (sanposhiho) [reply simulator/docs/external-scheduler.md:11]: Nah, probably the original one is better

**Comment by sanposhiho** [comment]
@196Ikuchil fixed 🙏

**Comment by sanposhiho** [comment]
rebased.

/unhold

File: simulator/docs/plugin-extender.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    e.handle.AddCustomResult(pod.Namespace, pod.Name, "noderesourcefit-prefilter-data", prefilterData)
+}
+```
+
+If you use the above extender, 
+each Pod will get `"noderesourcefit-prefilter-data": prefilterData` annotation in each scheduling like other scheduling results.
+
+### use plugin extender
+
+Currently, the plugin extender can be used only in [the external scheduler](./external-scheduler.md).
```
Comment (196Ikuchil) [inline simulator/docs/plugin-extender.md:49]:
```suggestion
**Currently, the plugin extender can be used only in [the external scheduler](./external-scheduler.md).**
```
Important 👍 
It should also be written at the top of the page.

**Review by 196Ikuchil — COMMENTED** [review]
@sanposhiho 
Thanks for your elaboration! Almost good to me.
I left some small nits. Plz check it.

File: README.md [hunk 6]
```diff
@@ -60,27 +109,14 @@ And, after pods are scheduled, you can see the results of
...
+(Note: changes to any fields other than `.profiles` are disabled on simulator, since they do not affect the results of the scheduling.)
 
 ![configure scheduler](simulator/docs/images/schedulerconfiguration.png)
 
-If you want to use your custom plugins as out-of-tree plugins in the simulator, please follow [this doc](simulator/docs/how-to-use-custom-plugins/README.md).
-
 ## Getting started
 
-You can find more information about environment variables available in the simulator server
-[here.](./simulator/docs/environment-variables.md)
```
[the remark this answers is not in the document]
Comment (sanposhiho) [reply README.md:78]: Yes, intended. 
We only need to tell users about the configuration via the config file since we deprecate the config via env vars.

**Comment by sanposhiho** [comment]
@196Ikuchil done 🙏

**Pull request merged**